File: cargo/src/bin/number/number.rs
use numbers_lib::{factorial, fibonacci};
use std::io::{self, Write};

fn get_user_number() -> i32 {
    print!("Enter a number : ");
    io::stdout().flush().unwrap();
    let mut number_str = String::new();
    io::stdin().read_line(&mut number_str).unwrap();
    number_str.trim().parse().unwrap_or(0)
}

pub fn print_number_calculations() {
    let number = get_user_number();
    println!("Fib[{}] = {}", number, fibonacci(number as usize));
    println!("Fact[{}] = {}", number, factorial(number as usize));
}

File: cabal/src/CLI/Function/Number.hs
module CLI.Function.Number (printNumberCalculations) where

import Numbers.Utils (factorial, fibonacci)
import System.IO (hFlush, stdout)
import Text.Read (readMaybe)

getUserNumber :: IO Integer
getUserNumber = do
  putStr "Enter a number : "
  hFlush stdout
  maybeNum <- fmap readMaybe getLine
  case maybeNum of
    Just num -> return num
    Nothing -> return 0

printNumberCalculations :: IO ()
printNumberCalculations = do
  number <- getUserNumber
  putStrLn $ "Fib[" ++ show number ++ "] = " ++ show (fibonacci number)
  putStrLn $ "Fact[" ++ show number ++ "] = " ++ show (factorial number)
